### DISCUSSION TITLE
Issues with varchars primary keys

### DISCUSSION THREAD

**Discussion by jadersbr** [filamentphp/filament#5347] [Help] [answered]
In my assets table, the id field is the primary key, however, it is not an auto-increment, it is a varchar(10).

Filament's datatable does not recognize this column and always returns the value 0. This same behavior happens with forms.

How can I resolve this?

Filament
![image](https://user-images.githubusercontent.com/58928002/209588956-f36088e2-37da-404a-b325-4211acecf2e9.png)

MySQL database
![image](https://user-images.githubusercontent.com/58928002/209588991-8c04540c-e2ea-48db-9964-154b8b215c51.png)

**Comment by ManojKiranA** [comment, marked as answer]
have you done changes to ur model

```
<?php

class Model extends Model
{
    protected $primaryKey = 'id';
    public $incrementing = false;
    protected $keyType = 'string';
}
```

**Reply by jadersbr** [reply]
Oh, I forgot that important detail. Thank you.
